client/jmx-adapter/src/test/java/org/jolokia/client/jmxadapter/JmxBridgeTest.java

```diff
@@ -242,19 +242,44 @@ public static Object[][] platformMBeans() {
       return testData.toArray(new Object[0][0]);
     } catch (Exception ignore) {
       return new Object[][]{
-          {ManagementFactory.CLASS_LOADING_MXBEAN_NAME, ClassLoadingMXBean.class},
-          {ManagementFactory.COMPILATION_MXBEAN_NAME, CompilationMXBean.class},
-          {ManagementFactory.MEMORY_MXBEAN_NAME, MemoryMXBean.class},
-          {ManagementFactory.OPERATING_SYSTEM_MXBEAN_NAME, OperatingSystemMXBean.class},
-          {ManagementFactory.RUNTIME_MXBEAN_NAME, RuntimeMXBean.class},
-          {ManagementFactory.THREAD_MXBEAN_NAME, ThreadMXBean.class},
+          {new ObjectName(ManagementFactory.CLASS_LOADING_MXBEAN_NAME), ClassLoadingMXBean.class},
+          {new ObjectName(ManagementFactory.COMPILATION_MXBEAN_NAME), CompilationMXBean.class},
+          {new ObjectName(ManagementFactory.MEMORY_MXBEAN_NAME), MemoryMXBean.class},
+          {new ObjectName(ManagementFactory.OPERATING_SYSTEM_MXBEAN_NAME), OperatingSystemMXBean.class},
+          {new ObjectName(ManagementFactory.RUNTIME_MXBEAN_NAME), RuntimeMXBean.class},
+          {new ObjectName(ManagementFactory.THREAD_MXBEAN_NAME), ThreadMXBean.class},
       };
     }
   }
 
 
   @DataProvider
   public static Object[][] safeOperationsToCall() {
+    String version = System.getProperty("java.specification.version");
+    if (version.contains(".")) {
+      version = version.substring(version.lastIndexOf('.') + 1);
+    }
+    int v = Integer.parseInt(version);
+    if (v <= 6) {
```

Comment:
This trick seems to be used in different places, so in the future it might be worth moving to a test helper class so other tests can share the code.

Reply:
ah yes - I decided to do it when there's a need to use it in 3rd place ;)

Reply:
good old criteria -- third time you repeat yourself, abstract it to shared code ;-)